storage/pebble/lookup.go

```diff
@@ -0,0 +1,88 @@
+package pebble
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/storage/pebble/registers/config"
+)
+
+// lookupKey is the encoded format of the storage key for looking up register value
+type lookupKey struct {
+	encoded []byte
+}
+
+// newLookupKey takes a height and registerID, returns the key for storing the register value in storage
+func newLookupKey(height uint64, reg flow.RegisterID) *lookupKey {
+	// Todo: Add prefixes for extensibility
+	key := lookupKey{
+		encoded: make([]byte, 0, len(reg.Owner)+1+len(reg.Key)+1+config.HeightSuffixLen),
+	}
+
+	// The lookup key used to find most recent value for a register.
+	//
+	// The "<owner>/<key>" part is the register key, which is used as a prefix to filter and iterate
+	// through updated values at different heights, and find the most recent updated value at or below
+	// a certain height.
+	key.encoded = append(key.encoded, []byte(reg.Owner)...)
+	key.encoded = append(key.encoded, '/')
+	key.encoded = append(key.encoded, []byte(reg.Key)...)
+	key.encoded = append(key.encoded, '/')
+
+	// Encode the height getting it to 1s compliment (all bits flipped) and big-endian byte order.
+	// (Prefix iteration in pebble does not support reverse iteration.)
```

Comment:
It took me a little bit to decode why we needed to use 1s complement here, so I write up some notes. Does this capture the rational?
```suggestion
	// Encode the height getting it to 1s compliment (all bits flipped) and big-endian byte order.
	//
	// Register are a sparse dataset stored with a single entry per update. To find the value at a particular
	// height, we need to do a scan across the entries to find the highest height that is less than or equal
	// to the target height.
	//
	// Pebble does not support reverse iteration, so we use the height's one's complement to effectively
	// reverse sort on the height. This allows us to use a bitwise forward scan for the next most recent
	// entry.
```